tools/cldr-code/src/main/java/org/unicode/cldr/util/StandardCodes.java

```diff
@@ -1025,6 +1042,27 @@ public String toCompatString() {
             }
         }
 
+        public NameType toNameType() {
+            switch (this) {
+                case region:
+                    return NameType.TERRITORY;
+                case language:
+                case legacy:
+                case redundant:
+                    return NameType.LANGUAGE;
+                case script:
+                    return NameType.SCRIPT;
+                case variant:
+                    return NameType.VARIANT;
+                case currency:
+                    return NameType.CURRENCY;
+                case subdivision:
+                    return NameType.SUBDIVISION;
+                default:
+                    return NameType.NONE;
+            }
+        }
+
```

Comment:
Similarly this maps StandardCodes.LstrType to NameType. I'm not sure how StandardCodes.LstrType and StandardCodes.CodeType are related; again, they seem to overlap